### PR TITLE
remove Ingest's Internal Template Service

### PR DESCRIPTION
**Pull request by talevy**
Ingest was using it's own wrapper around TemplateScripts and the ScriptService.
This commit removes that abstraction

### CODE REVIEW

**Comment by talevy** [comment]
jenkins, please test this

File: core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java [hunk 1]
```diff
@@ -265,10 +271,15 @@ public static ElasticsearchException newConfigurationException(String processorT
...
-                                                           String propertyValue, TemplateService templateService) {
+    public static TemplateScript.Factory compileTemplate(String processorType, String processorTag, String propertyName,
+                                                           String propertyValue, ScriptService scriptService) {
         try {
-            return templateService.compile(propertyValue);
+            if (scriptService.isLangSupported(DEFAULT_TEMPLATE_LANG)) {
+                Script script = new Script(ScriptType.INLINE, DEFAULT_TEMPLATE_LANG, propertyValue, Collections.emptyMap());
+                return scriptService.compile(script, TemplateScript.CONTEXT);
+            } else {
+                return (params) -> () -> propertyValue;
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java]: I don't like this but I understand why it is here. Can you please add a comment explaining that this is for rest tests when using the integ-test-zip?

**Review by rjernst — APPROVED** [review]
Thanks for this cleanup! LGTM, one small request.

**Review by martijnvg — APPROVED** [review]
Looks good! But I think we can now also remove `TemplateService` and `InternalTemplateService` classes because ingest no longer uses it?

**Comment by talevy** [comment]
thanks for the reviews @rjernst & @martijnvg!

**Comment by talevy** [comment]
jenkins, test this please

**Comment by talevy** [comment]
jenkins, test this please

**Comment by talevy** [comment]
jenkins, test this please

**Pull request merged**